data-prepper-plugins/http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/certificate/CertificateProviderFactory.java

```diff
@@ -17,13 +17,16 @@
 import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
 import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
 import software.amazon.awssdk.core.retry.RetryMode;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
 import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.services.acm.AcmClient;
 import software.amazon.awssdk.services.s3.S3Client;
 
 public class CertificateProviderFactory {
     private static final Logger LOG = LoggerFactory.getLogger(CertificateProviderFactory.class);
 
+    private final ApacheHttpClient.Builder apacheHttpClientBuilder = ApacheHttpClient.builder();
```

Comment:
Is creating a class level `ApacheHttpClient.Builder` the right approach? This may work for now given how simple this class is but the `ApacheHttpClient.Builder` maintains state . I am not sure if it clears the state after it builds a client or the status is preserved.

I would recommend limiting the scope of this variable by creating a new `Builder` for every `getCertificateProvider` call.

Reply:
I assumed that this wouldn't be as issue because we either create S3 or ACM client for every `getCertificateProvider` call. But I can make his change to limit the scope of it.

Reply:
Thanks, your fix ensures this won't be an issue in the future.